internal/translation/datafederation/conversion_endpoints.go

```diff
@@ -58,3 +65,39 @@ func endpointToAtlas(ep *DatafederationPrivateEndpointEntry) *admin.PrivateNetwo
 		Type:       pointer.MakePtrOrNil(ep.Type),
 	}
 }
+
+type DataFederationPrivateEndpoint struct {
+	AKO, Atlas, LastApplied *DatafederationPrivateEndpointEntry
+}
+
+func MapDatafederationPrivateEndpoints(projectID string, df *akov2.AtlasDataFederation, atlasEndpoints []*DatafederationPrivateEndpointEntry) (map[string]*DataFederationPrivateEndpoint, error) {
+	var lastApplied akov2.AtlasDataFederation
+	if ann, ok := df.GetAnnotations()[customresource.AnnotationLastAppliedConfiguration]; ok {
+		err := json.Unmarshal([]byte(ann), &lastApplied.Spec)
+		if err != nil {
+			return nil, fmt.Errorf("error reading data federation from last applied annotation: %w", err)
+		}
+	}
+
+	result := make(map[string]*DataFederationPrivateEndpoint)
+	addEndpoint := func(endpointID string) {
+		if _, ok := result[endpointID]; !ok {
+			result[endpointID] = &DataFederationPrivateEndpoint{}
+		}
+	}
```

Comment:
Why was a closure preferred to a simple external private function such as this?
```go
func(result map[string]*DataFederationPrivateEndpoint, endpointID string) {
	if _, ok := result[endpointID]; !ok {
		result[endpointID] = &DataFederationPrivateEndpoint{}
	}
}
```


Closures are required when the closure needs to travel to another context, bringing its context with it. This is not the case.

Reply:
This closes over `result`